### PR TITLE
Refactor appointments folder

### PR DESCRIPTION
**Pull request by Skn0tt**
This PR contains some refactorings to code in the `appointments` folder.

### CODE REVIEW

**Comment by Skn0tt** [comment]
I recommend reviewing commit-by-commit.

File: app/appointments/queries/getTimeSlots.ts [hunk 1]
```diff
@@ -1,28 +1,23 @@
...
 
-// Dummy data
-const dailySchedule = [
-  { day: "monday", startTime: "9:00", endTime: "17:00", meetingId: 2 },
-  { day: "tuesday", startTime: "9:00", endTime: "17:00", meetingId: 2 },
-  { day: "wednesday", startTime: "9:00", endTime: "17:00", meetingId: 2 },
-  { day: "thursday", startTime: "9:00", endTime: "17:00", meetingId: 2 },
-  { day: "friday", startTime: "9:00", endTime: "17:00", meetingId: 2 },
-]
+const mockNineToFiveSchedule: Schedule = {
```
Comment (rohansaw) [inline app/appointments/queries/getTimeSlots.ts:14]: why is this needed? I don't think we need this dummy data in production code :thinking:

File: app/appointments/utils/computeAvailableSlots.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  const endOfSearch = new Date(+between.end - durationInMilliseconds)
+
+  while (cursor < endOfSearch) {
+    const potentialSlot: TimeSlot = {
+      start: cursor,
+      end: new Date(+cursor + durationInMilliseconds),
+    }
+
+    const collidingSlot = takenSlots.find((slot) => collides(slot, potentialSlot))
```
Comment (rohansaw) [inline app/appointments/utils/computeAvailableSlots.ts:35]: This leads to timecomplexity of O(possibleSlots * Externalevents) which seems quite a lot to me. Thats why I sorted the externalEvents and only compared a possibleSlot  to the first event which would be sufficient an than removed it

Comment (rohansaw) [reply app/appointments/utils/computeAvailableSlots.ts:35]: But maybe the time-complexity also isn't so important here :shrug:

File: app/appointments/queries/getTimeSlots.ts (hunk 1 above)
Comment (Skn0tt) [reply app/appointments/queries/getTimeSlots.ts:14]: I don't know, you tell me 😅 It was there before and I figured that it isn't yet fetched from the database. Feel free to replace.

File: app/appointments/utils/computeAvailableSlots.ts (hunk 2 above)
Comment (Skn0tt) [reply app/appointments/utils/computeAvailableSlots.ts:35]: > This leads to timecomplexity of O(possibleSlots * Externalevents) which seems quite a lot to me.

Correct. I wanted to have a quick implementation down so I could add a test for it. Now that there's a test, improving efficiency is another issue.

> But maybe the time-complexity also isn't so important here 🤷

Yup, this is handling rather small problem instances anyway. If we want to improve responsiveness, we should better work on reducing our database calls :D

File: app/appointments/queries/getTimeSlots.ts (hunk 1 above)
Comment (rohansaw) [reply app/appointments/queries/getTimeSlots.ts:14]: Hmm weird, then maybe this was wrongly merged once, as I already replaced this in an earlier version :thinking: 
Will fix this tomorrow

File: app/appointments/utils/computeAvailableSlots.ts (hunk 2 above)
Comment (rohansaw) [reply app/appointments/utils/computeAvailableSlots.ts:35]: Yeah ok, I think we don't need to focus on this at the moment this is something that could be done in the future :)

**Comment by rohansaw** [comment]
In a seperate branch i now added usage of real schedule, but this somehow throws an `Maximum call size exceeded` Error.
The schedule is in the proper format so I think the error is caused by the refined scheduling algo. Maybe you could take a look at this @Skn0tt 
#
https://github.com/kalle-app/kalle/tree/use-real-schedule

**Comment by Skn0tt** [comment]
Will do tomorrow.

**Comment by Skn0tt** [comment]
> Will do tomorrow.

Actually, will do today.

**Comment by Skn0tt** [comment]
Turns out: using `>=` instead of `>` can fuck up your recursion 😅 9b73a480d514ad19a0756117e1d6d1891aade503

**Review by rohansaw — APPROVED** [review]

**Comment by Lasklu** [comment]
Can we delete the branch? @Skn0tt @rohansaw
